src/index.js

```diff
@@ -33,14 +33,12 @@ export default class Server extends EventEmitter {
   }
 
   handleRequest(req, res) {
-    const { path } = this;
     setState(res, states.CONNECTING);
-
-    if (!isValidPath(path, req)) {
-      return this.closeConnection(res, 400);
+    const headers = req.getAllHeaders();
+    if (/websocket/i.test(headers.get('Upgrade')) && /upgrade/i.test(headers.get('Connection'))) {
+      return this.handleUpgrade(req, res);
```

Comment:
@devsnek did you test this? From my understanding which may be wrong, this will handle stuff coming from `turbo-http` and in turn `turbo-net`. The `socket` provided by `turbo-net` is not compatible with the Node.js `net.Socket` so I _think_ you can't simply use an external server, unless that server is a `turbo-http` server. Is this the goal? It's not clear from the title.

Reply:
yes this only works with a turbo-http server instance

Reply:
Gotcha.

Reply:
Can you change those regex to `String.prototype.includes()`? Otherwise, I agree with @devsnek, because as long as you can attach event listeners and you follow the `turbo-http` API it should be fine.

Reply:
regex test allowed the values to not be there or be weirdly cased. otherwise I have to make sure they are strings and such

Reply:
if we have to worry about the case, i think the current solution is the fastest? https://jsperf.com/includes-vs-test/1